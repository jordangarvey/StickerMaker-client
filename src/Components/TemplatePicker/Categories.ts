/**
 * Enum defining the categories that a sticker can be edited by
 */
enum Categories {
	Colour = "colour",
	Shape = "shape",
	Text = "text"
}

export default Categories;
