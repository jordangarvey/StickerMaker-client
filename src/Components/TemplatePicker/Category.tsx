import React, { FC } from "react";
import { StyleSheet, Text, TouchableOpacity } from "react-native";

import Categories from "./Categories";

import * as C from "../../Global/Colours";


/**
 * Interface for the Category props
 */
interface ICategoryProps {
	/** The currently selected category this option belongs to e.g. shapes, text */
	category: Categories;
	/** On press callback to be fired whenever the option button is pressed */
	onPress(category: Categories): void;
	/** Optional boolean for whether or not this option is currently selected */
	selected?: boolean;
}

/**
 * Component to render a Category button
 */
const Category: FC<ICategoryProps> = (props) => {
	return (
		<TouchableOpacity
			style={[
				styles.category,
				props.selected && { backgroundColor: C.primary }
			]}
			onPress={props.onPress.bind(null, props.category)}
		>
			<Text style={props.selected && { color: C.secondary }}>{props.category.charAt(0).toUpperCase() + props.category.slice(1)}</Text>
		</TouchableOpacity>
	);
};

/**
 * Styles for the Category button
 */
const styles = StyleSheet.create({
	category: {
		alignItems: "flex-end",
		backgroundColor: C.secondary,
		borderRadius: 6,
		height: 80,
		flexDirection: "row",
		justifyContent: "center",
		marginHorizontal: 20,
		padding: 5,
		width: 60
	}
});

export default Category;
