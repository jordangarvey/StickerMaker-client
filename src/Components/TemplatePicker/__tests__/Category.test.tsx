import React from "react";
import renderer from "react-test-renderer";

import Categories from "../Categories";

import Category from "../Category";


describe("The Category componentâ€™s snapshot", () => {
	it("renders as expected", () => {
		function mockFunction() {
			console.log("Test")
		}

		const component = renderer.create(
			<Category
				category={Categories.Colour}
				onPress={mockFunction}
			/>
		).toJSON();


		expect(component).toMatchSnapshot();
	});
});
