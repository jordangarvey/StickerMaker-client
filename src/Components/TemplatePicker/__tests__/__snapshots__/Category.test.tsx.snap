// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Category componentâ€™s snapshot renders as expected 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "flex-end",
      "backgroundColor": "#FFFFFF",
      "borderRadius": 6,
      "flexDirection": "row",
      "height": 80,
      "justifyContent": "center",
      "marginHorizontal": 20,
      "opacity": 1,
      "padding": 5,
      "width": 60,
    }
  }
>
  <Text>
    Colour
  </Text>
</View>
`;
